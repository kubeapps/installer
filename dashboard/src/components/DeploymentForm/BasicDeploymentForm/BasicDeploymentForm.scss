.description {
  color: #5f6369;
  font-size: 0.9em;
}

.react-switch {
  vertical-align: middle;
  margin-left: 10px;
}

.disk_size_input {
  width: 75%;
}
