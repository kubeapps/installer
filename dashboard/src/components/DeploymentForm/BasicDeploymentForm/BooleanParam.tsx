import * as React from "react";
import Switch from "react-switch";
import { IBasicFormParam } from "shared/types";

export interface IStringParamProps {
  id: string;
  name: string;
  label: string;
  param: IBasicFormParam;
  handleBasicFormParamChange: (
    name: string,
    p: IBasicFormParam,
  ) => (e: React.FormEvent<HTMLInputElement>) => void;
}

class BooleanParam extends React.Component<IStringParamProps> {
  public render() {
    const { id, param, label } = this.props;
    return (
      <label htmlFor={id}>
        <div className="margin-b-normal">
          <span>{label}</span>
          <Switch
            id={id}
            onChange={this.handleChange}
            checked={param.value}
            className="react-switch"
          />
        </div>
        {param.description && <span className="description">{param.description}</span>}
      </label>
    );
  }

  // handleChange transform the event received by the Switch component to a checkbox event
  public handleChange = (checked: boolean) => {
    const { name, param } = this.props;
    const event = {
      currentTarget: { value: String(checked), type: "checkbox", checked },
    } as React.FormEvent<HTMLInputElement>;
    this.props.handleBasicFormParamChange(name, param)(event);
  };
}

export default BooleanParam;
