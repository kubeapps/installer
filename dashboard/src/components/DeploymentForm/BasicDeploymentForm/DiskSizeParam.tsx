import * as React from "react";
import { IBasicFormParam } from "shared/types";
import Slider from "../../../components/Slider";

export interface IDiskSizeParamProps {
  id: string;
  name: string;
  label: string;
  param: IBasicFormParam;
  handleBasicFormParamChange: (
    name: string,
    p: IBasicFormParam,
  ) => (e: React.FormEvent<HTMLInputElement>) => void;
}

export interface IDiskSizeParamState {
  Gi: number;
}

function toNumber(value: string) {
  // Force to return a Number from a string removing any character that is not a digit
  return Number(value.replace(/[^\d\.]/g, ""));
}

class DiskSizeParam extends React.Component<IDiskSizeParamProps, IDiskSizeParamState> {
  public state: IDiskSizeParamState = {
    Gi: toNumber(this.props.param.value) || 10,
  };

  // onChangeSlider is executed when the slider is dropped at one point
  // at that point we update the parameter
  public onChangeSlider = (values: number[]) => {
    this.handleParamChange(values[0]);
  };

  // onUpdateSlider is executed when dragging the slider
  // we just update the state here for a faster response
  public onUpdateSlider = (values: number[]) => {
    this.setState({ Gi: values[0] });
  };

  public onChangeInput = (e: React.FormEvent<HTMLInputElement>) => {
    const value = toNumber(e.currentTarget.value);
    this.setState({ Gi: value });
    this.handleParamChange(value);
  };

  public render() {
    const { param, label } = this.props;
    return (
      <div>
        <label htmlFor={this.props.id}>
          {label}
          {param.description && (
            <>
              <br />
              <span className="description">{param.description}</span>
            </>
          )}
          <div className="row">
            <div className="col-10">
              <Slider
                min={1}
                max={Math.max(100, this.state.Gi)}
                default={this.state.Gi}
                onChange={this.onChangeSlider}
                onUpdate={this.onUpdateSlider}
                values={this.state.Gi}
              />
            </div>
            <div className="col-2">
              <input
                className="disk_size_input"
                id={this.props.id}
                onChange={this.onChangeInput}
                value={this.state.Gi}
              />
              <span className="margin-l-normal">Gi</span>
            </div>
          </div>
        </label>
      </div>
    );
  }

  private handleParamChange = (value: number) => {
    this.props.handleBasicFormParamChange(this.props.name, this.props.param)({
      currentTarget: { value: `${value}Gi` },
    } as React.FormEvent<HTMLInputElement>);
  };
}

export default DiskSizeParam;
