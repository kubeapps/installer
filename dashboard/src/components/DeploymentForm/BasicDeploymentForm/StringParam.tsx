import * as React from "react";
import { IBasicFormParam } from "shared/types";

export interface IStringParamProps {
  id: string;
  name: string;
  label: string;
  param: IBasicFormParam;
  handleBasicFormParamChange: (
    name: string,
    p: IBasicFormParam,
  ) => (e: React.FormEvent<HTMLInputElement>) => void;
}

class StringParam extends React.Component<IStringParamProps> {
  public render() {
    const { id, name, param, label } = this.props;
    return (
      <div>
        <label htmlFor={id}>
          {label}
          {param.description && (
            <>
              <br />
              <span className="description">{param.description}</span>
            </>
          )}
          <input
            id={id}
            onChange={this.props.handleBasicFormParamChange(name, param)}
            value={param.value}
          />
        </label>
      </div>
    );
  }
}

export default StringParam;
