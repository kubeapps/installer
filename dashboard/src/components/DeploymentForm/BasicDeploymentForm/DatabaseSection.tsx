import * as React from "react";
import { setValue, USE_SELF_HOSTED_DB } from "../../../shared/schema";
import { IBasicFormParam } from "../../../shared/types";
import { getValueFromEvent } from "../../../shared/utils";
import BooleanParam from "./BooleanParam";
import TextParam from "./TextParam";

export interface IDatabaseSectionProps {
  label: string;
  param: IBasicFormParam;
  handleValuesChange: (value: string) => void;
  appValues: string;
}

class DatabaseSection extends React.Component<IDatabaseSectionProps> {
  public render() {
    const { label, param } = this.props;
    return (
      <div className="subsection margin-v-normal">
        <BooleanParam
          label="Use a Self Hosted Database"
          handleBasicFormParamChange={this.handleChildrenParamChange}
          id={"enable-self-hosted-db"}
          name={USE_SELF_HOSTED_DB}
          param={param.children![USE_SELF_HOSTED_DB]}
        />
        <div hidden={param.children![USE_SELF_HOSTED_DB].value} className="margin-t-normal">
          <span>{label}</span>
          {param.children &&
            Object.keys(param.children)
              .filter(p => p !== USE_SELF_HOSTED_DB)
              .map((paramName, i) => {
                return this.renderParam(paramName, param.children![paramName], i);
              })}
        </div>
      </div>
    );
  }

  private handleChildrenParamChange = (name: string, param: IBasicFormParam) => {
    return (e: React.FormEvent<HTMLInputElement>) => {
      const value = getValueFromEvent(e);
      this.props.handleValuesChange(setValue(this.props.appValues, param.path, value));
      this.props.param.children![name] = { ...this.props.param.children![name], value };
    };
  };

  private renderParam(name: string, param: IBasicFormParam, index: number) {
    const id = `${name}-${index}`;
    let label = "";
    let type = "text";
    switch (name) {
      case "externalDatabaseHost":
        label = "Host";
        break;
      case "externalDatabaseUser":
        label = "User";
        break;
      case "externalDatabasePassword":
        label = "Password";
        break;
      case "externalDatabaseName":
        label = "Database";
        break;
      case "externalDatabasePort":
        label = "Port";
        type = "number";
        break;
    }
    return (
      <TextParam
        label={label}
        handleBasicFormParamChange={this.handleChildrenParamChange}
        key={id}
        id={id}
        name={name}
        param={param}
        inputType={type}
      />
    );
  }
}

export default DatabaseSection;
