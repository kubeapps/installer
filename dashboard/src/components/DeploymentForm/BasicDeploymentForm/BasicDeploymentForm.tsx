import * as React from "react";
import { IBasicFormParam } from "shared/types";
import TextParam from "./TextParam";

import "./BasicDeploymentForm.css";
import BooleanParam from "./BooleanParam";
import DatabaseSection from "./DatabaseSection";
import DiskSizeParam from "./DiskSizeParam";

export interface IBasicDeploymentFormProps {
  params: { [name: string]: IBasicFormParam };
  handleBasicFormParamChange: (
    name: string,
    p: IBasicFormParam,
  ) => (e: React.FormEvent<HTMLInputElement>) => void;
  handleValuesChange: (value: string) => void;
  appValues: string;
}

class BasicDeploymentForm extends React.Component<IBasicDeploymentFormProps> {
  public render() {
    return Object.keys(this.props.params).map((paramName, i) => {
      return this.renderParam(paramName, this.props.params[paramName], i);
    });
  }

  private renderParam(name: string, param: IBasicFormParam, index: number) {
    const id = `${name}-${index}`;
    switch (name) {
      case "username":
        return (
          <TextParam
            label="Username"
            handleBasicFormParamChange={this.props.handleBasicFormParamChange}
            key={id}
            id={id}
            name={name}
            param={param}
          />
        );
      case "password":
        return (
          <TextParam
            label="Password"
            handleBasicFormParamChange={this.props.handleBasicFormParamChange}
            key={id}
            id={id}
            name={name}
            param={param}
          />
        );
      case "email":
        return (
          <TextParam
            label="Email"
            handleBasicFormParamChange={this.props.handleBasicFormParamChange}
            key={id}
            id={id}
            name={name}
            param={param}
          />
        );
      case "externalDatabase":
        return (
          <DatabaseSection
            label="External Database Details"
            handleValuesChange={this.props.handleValuesChange}
            appValues={this.props.appValues}
            key={id}
            param={param}
          />
        );
      case "diskSize":
        return (
          <DiskSizeParam
            label="Disk Size"
            handleBasicFormParamChange={this.props.handleBasicFormParamChange}
            key={id}
            id={id}
            name={name}
            param={param}
          />
        );
      default:
        switch (param.type) {
          case "string":
            return (
              <TextParam
                label={param.title || ""}
                handleBasicFormParamChange={this.props.handleBasicFormParamChange}
                key={id}
                id={id}
                name={name}
                param={param}
              />
            );
          case "integer":
            return (
              <TextParam
                label={param.title || ""}
                handleBasicFormParamChange={this.props.handleBasicFormParamChange}
                key={id}
                id={id}
                name={name}
                param={param}
                inputType="number"
              />
            );
          case "boolean":
            return (
              <BooleanParam
                label={param.title || ""}
                handleBasicFormParamChange={this.props.handleBasicFormParamChange}
                key={id}
                id={id}
                name={name}
                param={param}
              />
            );
          default:
          // TODO(andres): This should return an error once we add support for all the parameters that we expect
          // throw new Error(`Param ${name} with type ${param.type} is not supported`);
        }
    }
    return null;
  }
}

export default BasicDeploymentForm;
