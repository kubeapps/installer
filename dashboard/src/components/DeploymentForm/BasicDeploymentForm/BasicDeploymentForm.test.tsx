import * as React from "react";

import { mount } from "enzyme";
import { IBasicFormParam } from "shared/types";
import BasicDeploymentForm from "./BasicDeploymentForm";
import DatabaseSection from "./DatabaseSection";

const defaultProps = {
  params: [],
  handleBasicFormParamChange: jest.fn(() => jest.fn()),
  appValues: "",
  handleValuesChange: jest.fn(),
};

[
  {
    description: "renders a basic deployment with a username",
    params: { username: { path: "wordpressUsername", value: "user" } as IBasicFormParam },
  },
  {
    description: "renders a basic deployment with a password",
    params: {
      password: { path: "wordpressPassword", value: "sserpdrow" } as IBasicFormParam,
    },
  },
  {
    description: "renders a basic deployment with a email",
    params: { email: { path: "wordpressEmail", value: "user@example.com" } as IBasicFormParam },
  },
  {
    description: "renders a basic deployment with a generic string",
    params: {
      blogName: { path: "blogName", value: "my-blog", type: "string" } as IBasicFormParam,
    },
  },
  {
    description: "renders a basic deployment with a disk size",
    params: {
      diskSize: { path: "size", value: "10Gi", type: "string" } as IBasicFormParam,
    },
  },
  {
    description: "renders a basic deployment with username, password, email and a generic string",
    params: {
      username: { path: "wordpressUsername", value: "user" } as IBasicFormParam,
      password: { path: "wordpressPassword", value: "sserpdrow" } as IBasicFormParam,
      email: { path: "wordpressEmail", value: "user@example.com" } as IBasicFormParam,
      blogName: { path: "blogName", value: "my-blog", type: "string" } as IBasicFormParam,
    },
  },
  {
    description: "renders a basic deployment with a generic boolean",
    params: {
      enableMetrics: { path: "enableMetrics", value: true, type: "boolean" } as IBasicFormParam,
    },
  },
  {
    description: "renders a basic deployment with a generic number",
    params: {
      replicas: { path: "replicas", value: 1, type: "integer" } as IBasicFormParam,
    },
  },
].forEach(t => {
  it(t.description, () => {
    const onChange = jest.fn();
    const handleBasicFormParamChange = jest.fn(() => onChange);
    const wrapper = mount(
      <BasicDeploymentForm
        {...defaultProps}
        params={t.params as any}
        handleBasicFormParamChange={handleBasicFormParamChange}
      />,
    );
    expect(wrapper).toMatchSnapshot();
    Object.keys(t.params).map((param, i) => {
      wrapper.find(`input#${param}-${i}`).simulate("change");
      const mockCalls = handleBasicFormParamChange.mock.calls;
      expect(mockCalls[i]).toEqual([param, t.params[param]]);
      expect(onChange.mock.calls.length).toBe(i + 1);
    });
  });
});

it("should render an external database section", () => {
  const params = {
    externalDatabase: {
      path: "edbs",
      value: {},
      type: "object",
      children: {
        useSelfHostedDatabase: { path: "mariadb.enabled", value: {}, type: "boolean" },
      },
    },
  };
  const wrapper = mount(<BasicDeploymentForm {...defaultProps} params={params} />);

  const dbsec = wrapper.find(DatabaseSection);
  expect(dbsec).toExist();
});
