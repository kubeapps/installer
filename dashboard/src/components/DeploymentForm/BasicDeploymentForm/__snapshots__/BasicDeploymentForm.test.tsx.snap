// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a basic deployment with a disk size 1`] = `
<BasicDeploymentForm
  appValues=""
  handleBasicFormParamChange={[Function]}
  handleValuesChange={[Function]}
  params={
    Object {
      "diskSize": Object {
        "path": "size",
        "type": "string",
        "value": "10Gi",
      },
    }
  }
>
  <SliderParam
    handleBasicFormParamChange={[Function]}
    id="diskSize-0"
    key="diskSize-0"
    label="Disk Size"
    max={100}
    min={1}
    name="diskSize"
    param={
      Object {
        "path": "size",
        "type": "string",
        "value": "10Gi",
      }
    }
    unit="Gi"
  >
    <div>
      <label
        htmlFor="diskSize-0"
      >
        Disk Size
        <div
          className="row"
        >
          <div
            className="col-10"
          >
            <Slider
              default={10}
              max={100}
              min={1}
              onChange={[Function]}
              onUpdate={[Function]}
              values={10}
            >
              <Slider
                component="div"
                disabled={false}
                domain={
                  Array [
                    1,
                    100,
                  ]
                }
                flatten={false}
                mode={1}
                onChange={[Function]}
                onSlideEnd={[Function]}
                onSlideStart={[Function]}
                onUpdate={[Function]}
                reversed={false}
                rootProps={Object {}}
                rootStyle={
                  Object {
                    "margin": "1em",
                    "position": "relative",
                    "width": "90%",
                  }
                }
                step={1}
                values={
                  Array [
                    10,
                  ]
                }
                vertical={false}
                warnOnChanges={false}
              >
                <div
                  style={
                    Object {
                      "margin": "1em",
                      "position": "relative",
                      "width": "90%",
                    }
                  }
                >
                  <Rail
                    activeHandleID={null}
                    emitKeyboard={[Function]}
                    emitMouse={[Function]}
                    emitTouch={[Function]}
                    getEventData={[Function]}
                    handles={
                      Array [
                        Object {
                          "id": "$$-0",
                          "percent": 9.090909090909092,
                          "value": 10,
                        },
                      ]
                    }
                    key=".0"
                    scale={
                      LinearScale {
                        "domain": Array [
                          1,
                          100,
                        ],
                        "interpolator": [Function],
                        "range": Array [
                          0,
                          100,
                        ],
                      }
                    }
                  >
                    <div
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                      style={
                        Object {
                          "backgroundColor": "rgb(155,155,155)",
                          "borderRadius": 7,
                          "cursor": "pointer",
                          "height": 14,
                          "position": "absolute",
                          "width": "100%",
                        }
                      }
                    />
                  </Rail>
                  <Handles
                    activeHandleID={null}
                    emitKeyboard={[Function]}
                    emitMouse={[Function]}
                    emitTouch={[Function]}
                    getEventData={[Function]}
                    handles={
                      Array [
                        Object {
                          "id": "$$-0",
                          "percent": 9.090909090909092,
                          "value": 10,
                        },
                      ]
                    }
                    key=".1"
                    scale={
                      LinearScale {
                        "domain": Array [
                          1,
                          100,
                        ],
                        "interpolator": [Function],
                        "range": Array [
                          0,
                          100,
                        ],
                      }
                    }
                  >
                    <div
                      className="slider-handles"
                    >
                      <Component
                        domain={
                          Array [
                            1,
                            100,
                          ]
                        }
                        getHandleProps={[Function]}
                        handle={
                          Object {
                            "id": "$$-0",
                            "percent": 9.090909090909092,
                            "value": 10,
                          }
                        }
                        key="$$-0"
                      >
                        <div
                          aria-valuemax={100}
                          aria-valuemin={1}
                          aria-valuenow={10}
                          onKeyDown={[Function]}
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                          role="slider"
                          style={
                            Object {
                              "backgroundColor": "#34568f",
                              "borderRadius": "50%",
                              "boxShadow": "1px 1px 1px 1px rgba(0, 0, 0, 0.2)",
                              "cursor": "pointer",
                              "height": 24,
                              "left": "9.090909090909092%",
                              "marginLeft": "-11px",
                              "marginTop": "-6px",
                              "position": "absolute",
                              "width": 24,
                              "zIndex": 2,
                            }
                          }
                        />
                      </Component>
                    </div>
                  </Handles>
                  <Tracks
                    activeHandleID={null}
                    emitKeyboard={[Function]}
                    emitMouse={[Function]}
                    emitTouch={[Function]}
                    getEventData={[Function]}
                    handles={
                      Array [
                        Object {
                          "id": "$$-0",
                          "percent": 9.090909090909092,
                          "value": 10,
                        },
                      ]
                    }
                    key=".2"
                    left={true}
                    right={false}
                    scale={
                      LinearScale {
                        "domain": Array [
                          1,
                          100,
                        ],
                        "interpolator": [Function],
                        "range": Array [
                          0,
                          100,
                        ],
                      }
                    }
                  >
                    <div
                      className="slider-tracks"
                    >
                      <Component
                        getTrackProps={[Function]}
                        key="$-$$-0"
                        source={
                          Object {
                            "id": "$",
                            "percent": 0,
                            "value": 1,
                          }
                        }
                        target={
                          Object {
                            "id": "$$-0",
                            "percent": 9.090909090909092,
                            "value": 10,
                          }
                        }
                      >
                        <div
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                          style={
                            Object {
                              "backgroundColor": "#7aa0c4",
                              "borderRadius": 7,
                              "cursor": "pointer",
                              "height": 14,
                              "left": "0%",
                              "position": "absolute",
                              "width": "9.090909090909092%",
                              "zIndex": 1,
                            }
                          }
                        />
                      </Component>
                    </div>
                  </Tracks>
                </div>
              </Slider>
            </Slider>
          </div>
          <div
            className="col-2"
          >
            <input
              className="disk_size_input"
              id="diskSize-0"
              onChange={[Function]}
              value={10}
            />
            <span
              className="margin-l-normal"
            >
              Gi
            </span>
          </div>
        </div>
      </label>
    </div>
  </SliderParam>
</BasicDeploymentForm>
`;

exports[`renders a basic deployment with a email 1`] = `
<BasicDeploymentForm
  appValues=""
  handleBasicFormParamChange={[Function]}
  handleValuesChange={[Function]}
  params={
    Object {
      "email": Object {
        "path": "wordpressEmail",
        "value": "user@example.com",
      },
    }
  }
>
  <TextParam
    handleBasicFormParamChange={[Function]}
    id="email-0"
    inputType="string"
    key="email-0"
    label="email"
    name="email"
    param={
      Object {
        "path": "wordpressEmail",
        "value": "user@example.com",
      }
    }
  >
    <div>
      <label
        htmlFor="email-0"
      >
        email
        <input
          id="email-0"
          onChange={[Function]}
          type="string"
          value="user@example.com"
        />
      </label>
    </div>
  </TextParam>
</BasicDeploymentForm>
`;

exports[`renders a basic deployment with a generic boolean 1`] = `
<BasicDeploymentForm
  appValues=""
  handleBasicFormParamChange={[Function]}
  handleValuesChange={[Function]}
  params={
    Object {
      "enableMetrics": Object {
        "path": "enableMetrics",
        "type": "boolean",
        "value": true,
      },
    }
  }
>
  <BooleanParam
    handleBasicFormParamChange={[Function]}
    id="enableMetrics-0"
    key="enableMetrics-0"
    label="enableMetrics"
    name="enableMetrics"
    param={
      Object {
        "path": "enableMetrics",
        "type": "boolean",
        "value": true,
      }
    }
  >
    <label
      htmlFor="enableMetrics-0"
    >
      <div
        className="margin-b-normal"
      >
        <span>
          enableMetrics
        </span>
        <ReactSwitch
          activeBoxShadow="0 0 2px 3px #3bf"
          boxShadow={null}
          checked={true}
          checkedIcon={
            <svg
              height="100%"
              style={
                Object {
                  "position": "absolute",
                  "top": 0,
                }
              }
              viewBox="-2 -5 17 21"
              width="100%"
            >
              <path
                d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                fill="#fff"
                fillRule="evenodd"
              />
            </svg>
          }
          className="react-switch"
          disabled={false}
          height={28}
          id="enableMetrics-0"
          offColor="#888"
          offHandleColor="#fff"
          onChange={[Function]}
          onColor="#080"
          onHandleColor="#fff"
          uncheckedIcon={
            <svg
              height="100%"
              style={
                Object {
                  "position": "absolute",
                  "top": 0,
                }
              }
              viewBox="-2 -5 14 20"
              width="100%"
            >
              <path
                d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                fill="#fff"
                fillRule="evenodd"
              />
            </svg>
          }
          width={56}
        >
          <div
            className="react-switch"
            style={
              Object {
                "MozTransition": "opacity 0.25s",
                "MozUserSelect": "none",
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "WebkitTransition": "opacity 0.25s",
                "WebkitUserSelect": "none",
                "borderRadius": 14,
                "direction": "ltr",
                "display": "inline-block",
                "msUserSelect": "none",
                "opacity": 1,
                "position": "relative",
                "textAlign": "left",
                "touchAction": "none",
                "transition": "opacity 0.25s",
                "userSelect": "none",
              }
            }
          >
            <div
              className="react-switch-bg"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "MozTransition": "background 0.25s",
                  "WebkitTransition": "background 0.25s",
                  "background": "#008800",
                  "borderRadius": 14,
                  "cursor": "pointer",
                  "height": 28,
                  "margin": 0,
                  "position": "relative",
                  "transition": "background 0.25s",
                  "width": 56,
                }
              }
            >
              <div
                style={
                  Object {
                    "MozTransition": "opacity 0.25s",
                    "WebkitTransition": "opacity 0.25s",
                    "height": 28,
                    "opacity": 1,
                    "pointerEvents": "none",
                    "position": "relative",
                    "transition": "opacity 0.25s",
                    "width": 30,
                  }
                }
              >
                <svg
                  height="100%"
                  style={
                    Object {
                      "position": "absolute",
                      "top": 0,
                    }
                  }
                  viewBox="-2 -5 17 21"
                  width="100%"
                >
                  <path
                    d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
                    fill="#fff"
                    fillRule="evenodd"
                  />
                </svg>
              </div>
              <div
                style={
                  Object {
                    "MozTransition": "opacity 0.25s",
                    "WebkitTransition": "opacity 0.25s",
                    "height": 28,
                    "opacity": 0,
                    "pointerEvents": "none",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transition": "opacity 0.25s",
                    "width": 30,
                  }
                }
              >
                <svg
                  height="100%"
                  style={
                    Object {
                      "position": "absolute",
                      "top": 0,
                    }
                  }
                  viewBox="-2 -5 14 20"
                  width="100%"
                >
                  <path
                    d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
                    fill="#fff"
                    fillRule="evenodd"
                  />
                </svg>
              </div>
            </div>
            <div
              className="react-switch-handle"
              onClick={[Function]}
              onMouseDown={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                  "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                  "background": "#ffffff",
                  "border": 0,
                  "borderRadius": "50%",
                  "boxShadow": null,
                  "cursor": "pointer",
                  "display": "inline-block",
                  "height": 26,
                  "outline": 0,
                  "position": "absolute",
                  "top": 1,
                  "transform": "translateX(29px)",
                  "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
                  "width": 26,
                }
              }
            />
            <input
              checked={true}
              disabled={false}
              id="enableMetrics-0"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyUp={[Function]}
              role="switch"
              style={
                Object {
                  "border": 0,
                  "clip": "rect(0 0 0 0)",
                  "height": 1,
                  "margin": -1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "absolute",
                  "width": 1,
                }
              }
              type="checkbox"
            />
          </div>
        </ReactSwitch>
      </div>
    </label>
  </BooleanParam>
</BasicDeploymentForm>
`;

exports[`renders a basic deployment with a generic number 1`] = `
<BasicDeploymentForm
  appValues=""
  handleBasicFormParamChange={[Function]}
  handleValuesChange={[Function]}
  params={
    Object {
      "replicas": Object {
        "path": "replicas",
        "type": "integer",
        "value": 1,
      },
    }
  }
>
  <TextParam
    handleBasicFormParamChange={[Function]}
    id="replicas-0"
    inputType="number"
    key="replicas-0"
    label="replicas"
    name="replicas"
    param={
      Object {
        "path": "replicas",
        "type": "integer",
        "value": 1,
      }
    }
  >
    <div>
      <label
        htmlFor="replicas-0"
      >
        replicas
        <input
          id="replicas-0"
          onChange={[Function]}
          type="number"
          value={1}
        />
      </label>
    </div>
  </TextParam>
</BasicDeploymentForm>
`;

exports[`renders a basic deployment with a generic string 1`] = `
<BasicDeploymentForm
  appValues=""
  handleBasicFormParamChange={[Function]}
  handleValuesChange={[Function]}
  params={
    Object {
      "blogName": Object {
        "path": "blogName",
        "type": "string",
        "value": "my-blog",
      },
    }
  }
>
  <TextParam
    handleBasicFormParamChange={[Function]}
    id="blogName-0"
    inputType="string"
    key="blogName-0"
    label="blogName"
    name="blogName"
    param={
      Object {
        "path": "blogName",
        "type": "string",
        "value": "my-blog",
      }
    }
  >
    <div>
      <label
        htmlFor="blogName-0"
      >
        blogName
        <input
          id="blogName-0"
          onChange={[Function]}
          type="string"
          value="my-blog"
        />
      </label>
    </div>
  </TextParam>
</BasicDeploymentForm>
`;

exports[`renders a basic deployment with a password 1`] = `
<BasicDeploymentForm
  appValues=""
  handleBasicFormParamChange={[Function]}
  handleValuesChange={[Function]}
  params={
    Object {
      "password": Object {
        "path": "wordpressPassword",
        "value": "sserpdrow",
      },
    }
  }
>
  <TextParam
    handleBasicFormParamChange={[Function]}
    id="password-0"
    inputType="string"
    key="password-0"
    label="password"
    name="password"
    param={
      Object {
        "path": "wordpressPassword",
        "value": "sserpdrow",
      }
    }
  >
    <div>
      <label
        htmlFor="password-0"
      >
        password
        <input
          id="password-0"
          onChange={[Function]}
          type="string"
          value="sserpdrow"
        />
      </label>
    </div>
  </TextParam>
</BasicDeploymentForm>
`;

exports[`renders a basic deployment with a username 1`] = `
<BasicDeploymentForm
  appValues=""
  handleBasicFormParamChange={[Function]}
  handleValuesChange={[Function]}
  params={
    Object {
      "username": Object {
        "path": "wordpressUsername",
        "value": "user",
      },
    }
  }
>
  <TextParam
    handleBasicFormParamChange={[Function]}
    id="username-0"
    inputType="string"
    key="username-0"
    label="username"
    name="username"
    param={
      Object {
        "path": "wordpressUsername",
        "value": "user",
      }
    }
  >
    <div>
      <label
        htmlFor="username-0"
      >
        username
        <input
          id="username-0"
          onChange={[Function]}
          type="string"
          value="user"
        />
      </label>
    </div>
  </TextParam>
</BasicDeploymentForm>
`;

exports[`renders a basic deployment with username, password, email and a generic string 1`] = `
<BasicDeploymentForm
  appValues=""
  handleBasicFormParamChange={[Function]}
  handleValuesChange={[Function]}
  params={
    Object {
      "blogName": Object {
        "path": "blogName",
        "type": "string",
        "value": "my-blog",
      },
      "email": Object {
        "path": "wordpressEmail",
        "value": "user@example.com",
      },
      "password": Object {
        "path": "wordpressPassword",
        "value": "sserpdrow",
      },
      "username": Object {
        "path": "wordpressUsername",
        "value": "user",
      },
    }
  }
>
  <TextParam
    handleBasicFormParamChange={[Function]}
    id="username-0"
    inputType="string"
    key="username-0"
    label="username"
    name="username"
    param={
      Object {
        "path": "wordpressUsername",
        "value": "user",
      }
    }
  >
    <div>
      <label
        htmlFor="username-0"
      >
        username
        <input
          id="username-0"
          onChange={[Function]}
          type="string"
          value="user"
        />
      </label>
    </div>
  </TextParam>
  <TextParam
    handleBasicFormParamChange={[Function]}
    id="password-1"
    inputType="string"
    key="password-1"
    label="password"
    name="password"
    param={
      Object {
        "path": "wordpressPassword",
        "value": "sserpdrow",
      }
    }
  >
    <div>
      <label
        htmlFor="password-1"
      >
        password
        <input
          id="password-1"
          onChange={[Function]}
          type="string"
          value="sserpdrow"
        />
      </label>
    </div>
  </TextParam>
  <TextParam
    handleBasicFormParamChange={[Function]}
    id="email-2"
    inputType="string"
    key="email-2"
    label="email"
    name="email"
    param={
      Object {
        "path": "wordpressEmail",
        "value": "user@example.com",
      }
    }
  >
    <div>
      <label
        htmlFor="email-2"
      >
        email
        <input
          id="email-2"
          onChange={[Function]}
          type="string"
          value="user@example.com"
        />
      </label>
    </div>
  </TextParam>
  <TextParam
    handleBasicFormParamChange={[Function]}
    id="blogName-3"
    inputType="string"
    key="blogName-3"
    label="blogName"
    name="blogName"
    param={
      Object {
        "path": "blogName",
        "type": "string",
        "value": "my-blog",
      }
    }
  >
    <div>
      <label
        htmlFor="blogName-3"
      >
        blogName
        <input
          id="blogName-3"
          onChange={[Function]}
          type="string"
          value="my-blog"
        />
      </label>
    </div>
  </TextParam>
</BasicDeploymentForm>
`;
